runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/config/StreamingEngineComputationConfigFetcherTest.java

```diff
@@ -0,0 +1,198 @@
+    Set<StreamingEnginePipelineConfig> receivedPipelineConfig = new HashSet<>();
+    when(mockDataflowServiceClient.getGlobalStreamingConfigWorkItem())
+        .thenReturn(Optional.of(firstConfig))
+        .thenReturn(Optional.of(secondConfig))
+        .thenReturn(Optional.of(thirdConfig));
+
+    streamingEngineConfigFetcher =
+        createConfigLoader(
+            /* waitForInitialConfig= */ true,
+            Duration.millis(100).getMillis(),
+            config -> {
+              receivedPipelineConfig.add(config);
+              numExpectedRefreshes.countDown();
+            });
+
+    Thread asyncStartConfigLoader = new Thread(streamingEngineConfigFetcher::start);
+    asyncStartConfigLoader.start();
+    numExpectedRefreshes.await();
+    asyncStartConfigLoader.join();
+    assertThat(receivedPipelineConfig)
+        .containsExactly(
+            StreamingEnginePipelineConfig.builder()
+                .setMaxWorkItemCommitBytes(
+                    firstConfig.getStreamingConfigTask().getMaxWorkItemCommitBytes())
+                .build(),
+            StreamingEnginePipelineConfig.builder()
+                .setMaxWorkItemCommitBytes(
+                    secondConfig.getStreamingConfigTask().getMaxWorkItemCommitBytes())
+                .build(),
+            StreamingEnginePipelineConfig.builder()
+                .setMaxWorkItemCommitBytes(
+                    thirdConfig.getStreamingConfigTask().getMaxWorkItemCommitBytes())
+                .build());
+  }
+
+  @Test
+  public void testGetComputationConfig() throws IOException {
+    streamingEngineConfigFetcher =
+        createConfigLoader(/* waitForInitialConfig= */ false, 0, ignored -> {});
+    String computationId = "computationId";
+    String stageName = "stageName";
+    String systemName = "systemName";
+    StreamingComputationConfig pipelineConfig =
+        new StreamingComputationConfig()
+            .setComputationId(computationId)
+            .setStageName(stageName)
+            .setSystemName(systemName)
+            .setInstructions(ImmutableList.of());
+
+    WorkItem workItem =
+        new WorkItem()
+            .setStreamingConfigTask(
+                new StreamingConfigTask()
+                    .setStreamingComputationConfigs(ImmutableList.of(pipelineConfig)));
+
+    when(mockDataflowServiceClient.getStreamingConfigWorkItem(anyString()))
+        .thenReturn(Optional.of(workItem));
+    Optional<ComputationConfig> actualPipelineConfig =
+        streamingEngineConfigFetcher.fetchConfig(computationId);
+
+    assertTrue(actualPipelineConfig.isPresent());
+    assertThat(actualPipelineConfig.get())
+        .isEqualTo(
+            ComputationConfig.create(
+                StreamingEngineComputationConfigFetcher.createMapTask(pipelineConfig),
+                ImmutableMap.of(),
+                ImmutableMap.of()));
+  }
+
+  @Test
+  public void testGetComputationConfig_noComputationPresent() throws IOException {
```

Comment:
test the dataflowserviceclient returnign an exception

Reply:
done